### PR TITLE
RELATED: RAIL-4753 Refactor InsightDrillPanel 

### PR DESCRIPTION
**Pull request by ivanmjartan**
JIRA: RAIL-4753

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10883/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12970/ : SUCCESS in 13m 06s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10884/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12971/ : SUCCESS in 14m 07s

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/InsightDrillConfigPanel/useInsightDrillConfigPanel.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+    const onSetupItem = useCallback(
+        (drill: InsightDrillDefinition, changedItem: IDrillConfigItem) => {
+            const isNew = isItemNew(changedItem);
+            dispatch(modifyDrillsForInsightWidget(widgetRef, [drill]));
+
+            // we are not able remove incomplete items directly,it will change items in panel while command is processing
+            // and this will not keep correct scroll.
+            // we mark item compleat and remove is done in useIncompleteItems when widget drills changed
+            compleatItem(changedItem);
```
[the remark this answers is not in the document]
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/InsightDrillConfigPanel/useInsightDrillConfigPanel.ts]: fiexd

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3337

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Pull request merged**